spk/vim/Makefile

```diff
@@ -20,6 +18,10 @@ CHANGELOG = "Update vim to v9.0<br/>2. Migrate to Python 3.11"
 HOMEPAGE = https://www.vim.org/
 LICENSE  = Charityware
 
+ifneq ($(findstring $(ARCH),$(ARMv5_ARCHS) $(OLD_PPC_ARCHS)),$(ARCH))
+SPK_DEPENDS = "python311>=3.11.5-8"
+endif
+
```

Comment:
Can't we keep this dependency optional and remove the SPK_DEPENDS?

So (as before) you can install vim without python, but you can't use the python feature until you install python311.
If one wants to use the python feature, now python311 is needed and can be installed before or after the vim package.

So the changelog must change `2. Migrate to Python 3.11` to `2. You now need Python 3.11 to enable Python support.`

Probably it is not enough to mention this in the changelog, and we should add a note (except for ARMv5 and OLD_PPC) to the package description(s) like:

"To enable Python support you need to install the Python 3.11 package" 


_sorry to remain pedantic, but why sould a user need to install python for the update, when the python feature is not needed?_

Reply:
This really was to demonstrate two things:
1. it's still possible to build `vim` for armv5
2. it's also possible to migrate `vim` to python 3.11

As far as python being optional, I had not looked into that just yet.  Indeed it can be but there is a need to provide proper messaging on that matter... And not sure the changlog is the best but certainly is the easiest (wizard perhaps would be better but more complex due to arch build differences?).  Personnally I find making it mandatory by default is the simplest and safest.

But AFAIK you seem to object to that therefore I can change this easily by switching `SPK_DEPENDS` to a changelog += instead.  Before I make the necessary changes, would that meet your train of thoughts?

Reply:
@hgy59 Would https://github.com/SynoCommunity/spksrc/pull/5866/commits/1a355eb58d607827e5c12ae800a32fc79921a1a3 + https://github.com/SynoCommunity/spksrc/pull/5866/commits/59b561d539bcd4e031f7e367e34720fa94099acb suit your need?  I've checked and the `INFO` looks OK from a `"` perspective:
```
$ cat work-88f6281-6.2.4/INFO 
package="vim"
version="9.0-6"
description="Vim is a highly configurable text editor built to enable efficient text editing. It is an improved version of the vi editor distributed with most UNIX systems."
description_csy="Vim je pokročilý textový editor, který usiluje o to poskytnout sílu de-facto Unixového editoru Vi, s komplexnější sadou vlastností."
description_fre="Vim est un éditeur de texte avancé basé sur Vi (unix) avec des fonctions plus complètes."

arch="88f6281"
maintainer="SynoCommunity"
maintainer_url="https://github.com/SynoCommunity"
distributor=""
distributor_url=""
os_min_ver="6.2-25556"
helpurl="https://www.vim.org/"
ctl_stop="no"
displayname="Vim"
dsmappname="com.synocommunity.packages.vim"
changelog="1. Update vim to v9.0"
checksum="2d40a3b4be313ac1f3f7737fb25b698b"
```

```
$ cat work-x64-7.1/INFO 
package="vim"
version="9.0-6"
description="Vim is a highly configurable text editor built to enable efficient text editing. It is an improved version of the vi editor distributed with most UNIX systems."
description_csy="Vim je pokročilý textový editor, který usiluje o to poskytnout sílu de-facto Unixového editoru Vi, s komplexnější sadou vlastností."
description_fre="Vim est un éditeur de texte avancé basé sur Vi (unix) avec des fonctions plus complètes."

arch="apollolake avoton braswell broadwell broadwellnk broadwellnkv2 broadwellntbap bromolow cedarview denverton epyc7002 geminilake grantley kvmx64 purley r1000 v1000"
maintainer="SynoCommunity"
maintainer_url="https://github.com/SynoCommunity"
distributor=""
distributor_url=""
os_min_ver="7.1-42661"
helpurl="https://www.vim.org/"
ctl_stop="no"
displayname="Vim"
dsmappname="com.synocommunity.packages.vim"
changelog="1. Update vim to v9.0 <br/>2. Migrate to Python 3.11 <br/><br/>NOTE: You need to install Python 3.11 package to enable Python support."
checksum="721d1b9df75c45c26927d4dcfef56b52"
```